### PR TITLE
fix(admin): hard delete users with FK cleanup

### PR DESCRIPTION
**Pull request by BreakableHoodie**
## Summary

- Changes admin user deletion from soft delete (`UPDATE users SET is_active = 0`) to hard delete (`DELETE FROM users`)
- Safely unlinks the user from all related tables before deletion to avoid FK constraint failures
- Uses `DB.batch()` for atomicity — if any cleanup step fails, the whole operation rolls back

## FK Cleanup Order

Before deleting the user row, the batch nulls out `created_by_user_id` / `updated_by_user_id` on:
- `events`
- `venues`
- `band_profiles`
- `performances`

Tables handled automatically by schema constraints:
- `lucia_sessions` → `ON DELETE CASCADE`
- `password_reset_tokens` → `ON DELETE CASCADE`
- `audit_log` / `invite_codes` → `ON DELETE SET NULL`

## Test plan

- [ ] Admin deletes a non-admin user — user row no longer exists in DB (not just `is_active = 0`)
- [ ] Events/venues/band_profiles/performances previously created by the deleted user still exist with `created_by_user_id = NULL`
- [ ] Deleted user's sessions are gone from `lucia_sessions`
- [ ] Unit test in `user-update-delete.test.js` asserts `SELECT * FROM users WHERE id = ?` returns `undefined`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: functions/api/admin/users/[id].js [hunk 1]
```diff
@@ -287,7 +287,7 @@ export async function onRequestPatch(context) {
   }
 }
 
-// DELETE - Soft delete user (admin only)
+// DELETE - Hard delete user (admin only)
```
[the remark this answers is not in the document]
Comment (BreakableHoodie) [reply functions/api/admin/users/[id].js:290]: The full FK cleanup implementation is in the PR — it's spread across multiple commits. The diff you're seeing is the last commit only (a stale comment fix). The full PR diff includes:

- **Commit 3c8bbb0**: Initial soft→hard delete (`UPDATE users SET is_active = 0` → `DELETE FROM users`)
- **Commit 8b6ed2f**: Full FK cleanup using `DB.batch()` — nulls `created_by_user_id`/`updated_by_user_id` on events, venues, band_profiles, performances before deleting the user row; relies on `ON DELETE CASCADE` for lucia_sessions/password_reset_tokens
- **Commit 8b6ed2f** also adds 51 lines of tests asserting the user row is `undefined` post-delete and FK-linked rows are preserved with `NULL`

You can see the full implementation in the [PR's Files Changed tab](https://github.com/BreakableHoodie/settimesdotca/pull/278/files) — it shows all commits combined.
